reddit_decider/__init__.py

```diff
@@ -826,40 +795,59 @@ def get_all_dynamic_configs(self) -> List[Dict[str, Any]]:
 
         :return: list of all active dynamic config dicts.
         """
-        decider = self._get_decider()
-        if not decider:
-            return []
-
-        ctx = self._get_ctx()
-        ctx_err = ctx.err()
-        if ctx_err is not None:
-            logger.info(f"Encountered error in rust_decider.make_ctx(): {ctx_err}")
+        if self._internal is None:
+            logger.error("rs_decider is None--did not initialize.")
             return []
 
-        all_decisions_result = decider.get_all_values(ctx)
+        ctx = self._decider_context.to_dict()
 
-        error = all_decisions_result.err()
-        if error:
-            logger.info(f"Encountered error in decider.choose_all(): {error}")
+        try:
+            values = self._internal.all_values(ctx)
+        except DeciderException as exc:
+            logger.info(str(exc))
             return []
 
-        all_decisions = all_decisions_result.decisions()
         parsed_configs = []
 
-        for dc_name, decision in all_decisions.items():
-            decision_error = decision.err()
-            if decision_error:
-                logger.info(
-                    f"Encountered error for dynamic config: {dc_name} in decider.get_all_values(): {decision_error}"
-                )
-                continue
+        for feature_name, val in values.items():
+            parsed_configs.append(self._value_to_dc_dict(feature_name, val))
 
-            value_dict = decision.value_dict()
+        return parsed_configs
 
-            if value_dict:
-                parsed_configs.append(value_dict)
+    def _get_dynamic_config_value(
+        self,
+        feature_name: str,
+        default: Any,
+        dc_types: List[type],
+    ) -> Any:
+        if self._internal is None:
+            logger.error("rs_decider is None--did not initialize.")
+            return default
 
-        return parsed_configs
+        ctx = self._decider_context.to_dict()
+
+        try:
+            decision = self._internal.choose(feature_name=feature_name, context=ctx)
+        except FeatureNotFoundException as exc:
+            warnings.warn(str(exc))
+            return default
+        except DeciderException as exc:
+            logger.info(str(exc))
+            return default
+
+        value = decision.value
+
+        if type(value) not in dc_types:
+            return default
+
+        return value
```

Comment:
If we have a single type above, we can do

```suggestion
        try:
            return dc_type(value)
        except TypeError:
            return default
```

Reply:
niceeee, added.